tools/block-generator/generator/generate.go

```diff
@@ -595,45 +494,44 @@ func getAppTxOptions() []interface{} {
 
 // ---- Transaction Generation (Pay/Asset/Apps) ----
 
-func (g *generator) generateSignedTxns(round uint64, intra uint64) ([]txn.SignedTxn, []txn.ApplyData, uint64 /* nextIntra */, error) {
+func (g *generator) generateTxGroup(round uint64, intra uint64) ([]txn.SignedTxnWithAD, uint64 /* nextIntra */, error) {
 	// TODO: return the number of transactions generated instead of updating intra!!!
 	selection, err := weightedSelection(g.transactionWeights, getTransactionOptions(), paymentTx)
 	if err != nil {
-		return nil, nil, intra, err
+		return nil, intra, err
 	}
 
 	var signedTxns []txn.SignedTxn
-	var ads []txn.ApplyData
 	var nextIntra uint64
 	var expectedID uint64
 	switch selection {
 	case paymentTx:
 		var signedTxn txn.SignedTxn
-		var ad txn.ApplyData
-		signedTxn, ad, nextIntra, err = g.generatePaymentTxn(round, intra)
+		signedTxn, nextIntra, err = g.generatePaymentTxn(round, intra)
```

Comment:
the `generateXYZtxn()` methods no longer return `ApplyData` (which were always empty)